test/unitTests/spec/visit/components/checkIn.spec.js

```diff
@@ -98,34 +98,75 @@ describe('CheckInController', function () {
         });
       });
 
+      var ctrl = $componentController('checkIn', null, {onCheckInChange: jasmine.createSpy('onCheckInChange')});
+
+      ctrl.checkIn();
+      $rootScope.$apply();
+
+      expect(ctrl.onCheckInChange).toHaveBeenCalled();
+
+    });
+
+  });
+
+  describe('cancelCheckIn', function () {
+
+    it('should check-in the patient', function () {
+
+      var visit = {};
+
+      spyOn(visitService, 'deleteVisit').and.callFake(function () {
+        return $q(function (resolve) {
+          return resolve(visit);
+        });
+      });
+
+      var ctrl = $componentController('checkIn');
+
+      ctrl.cancelCheckIn();
+
+      expect(visitService.deleteVisit).toHaveBeenCalled();
+
+    });
+
+    it('should update the day\'s visit', function () {
+
+      var visit = {};
+
+      spyOn(visitService, 'deleteVisit').and.callFake(function () {
+        return $q(function (resolve) {
+          return resolve(visit);
+        });
+      });
+
       var ctrl = $componentController('checkIn', null, {
-        onCheckIn: function () {
+        onCheckInChange: function () {
         }
       });
 
-      ctrl.checkIn();
+      ctrl.cancelCheckIn();
       $rootScope.$apply();
 
-      expect(ctrl.disableCheckin).toEqual(true);
+      expect(ctrl.todayVisit).toEqual(null);
 
     });
 
-    it('should call onCheckIn binding', function () {
+    it('should call onCheckInChange binding', function () {
 
       var visit = {};
 
-      spyOn(visitService, 'checkInPatient').and.callFake(function () {
+      spyOn(visitService, 'deleteVisit').and.callFake(function () {
         return $q(function (resolve) {
           return resolve(visit);
         });
       });
 
-      var ctrl = $componentController('checkIn', null, { onCheckIn: jasmine.createSpy('onCheckIn') });
+      var ctrl = $componentController('checkIn', null, {onCheckInChange: jasmine.createSpy('onCheckInChange')});
 
-      ctrl.checkIn();
+      ctrl.cancelCheckIn();
       $rootScope.$apply();
 
-      expect(ctrl.onCheckIn).toHaveBeenCalled();
+      expect(ctrl.onCheckInChange).toHaveBeenCalled();
```

Comment:
How about ```expect(ctrl.onCheckInChange).toHaveBeenCalledTimes(2)``` to make sure it's been called exactly twice?

Reply:
Is it supposed to be called twice?

Reply:
Np :). My bad. I thought ```checkIn``` was called before ```cancelCheckIn```